### PR TITLE
added http2 client to consumers

### PR DESCRIPTION
**Pull request by wojtkiewicz**
Added basic support for http2 delivery to hermes-consumer.

### CODE REVIEW

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java [hunk 1]
```diff
@@ -51,21 +63,40 @@ public MessageSender create(Subscription subscription) {
 
     private HttpRequestFactory httpRequestFactory(Subscription subscription) {
         int requestTimeout = subscription.getSerialSubscriptionPolicy().getRequestTimeout();
-        return new HttpRequestFactory(httpClient, requestTimeout, metadataAppender, authorizationProviderFactory.create(subscription));
+        HttpClient client = subscription.isHttp2Enabled() ? http2Client.orElseGet(() -> {
+            logger.info("Http2 delivery is not enabled on this server. Falling back to http1.");
+            return httpClient;
+        }) : httpClient;
+        return new HttpRequestFactory(http2Client.get(), requestTimeout, metadataAppender, authorizationProviderFactory.create(subscription));
```
Comment (dankraw) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: We need to remember about switching the http client for given subscription when the subscription is being updated (the `http2Enabled` property). This requires change in `ConsumerMessageSender.updateSubscription(subscription)`.

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java [hunk 2]
```diff
@@ -51,21 +63,40 @@ public MessageSender create(Subscription subscription) {
...
     }
 
 
     @Override
     public void start() throws Exception {
-        if (httpClient.isStopped()) {
-            httpClient.start();
+        startClient(httpClient);
+        if (http2Client.isPresent()) {
+            startClient(http2Client.get());
```
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: ...throws Exception

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java [hunk 3]
```diff
@@ -51,21 +63,40 @@ public MessageSender create(Subscription subscription) {
...
         }
     }
 
     @Override
     public void stop() throws Exception {
-        if (httpClient.isRunning()) {
-            httpClient.stop();
+        stopClient(httpClient);
+        if (http2Client.isPresent()) {
+            stopClient(http2Client.get());
```
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: ...throws Exception

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java [hunk 4]
```diff
@@ -51,21 +63,40 @@ public MessageSender create(Subscription subscription) {
 
     private HttpRequestFactory httpRequestFactory(Subscription subscription) {
         int requestTimeout = subscription.getSerialSubscriptionPolicy().getRequestTimeout();
-        return new HttpRequestFactory(httpClient, requestTimeout, metadataAppender, authorizationProviderFactory.create(subscription));
+        HttpClient client = subscription.isHttp2Enabled() ? http2Client.orElseGet(() -> {
+            logger.info("Http2 delivery is not enabled on this server. Falling back to http1.");
```
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: good idea

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java (hunk 1 above)
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: good idea

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java (hunk 3 above)
Comment (dankraw) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: thx for the info 👍

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java (hunk 2 above)
Comment (dankraw) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: alright 👍

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java (hunk 1 above)
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: done

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java (hunk 2 above)
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: done anyway

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java (hunk 3 above)
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: done anyway

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java (hunk 4 above)
Comment (wojtkiewicz) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/JettyHttpMessageSenderProvider.java]: done

**Pull request merged**